### PR TITLE
fix: uints, missing functions, match all function args, prettier

### PR DESCRIPTION
**Pull request by reedrosenbluth**
Fixes:

1. Highlight uints
2. Add missing functions (`stx-...`)
2. Match all function args, not just the last one
3. Prettier formatting

### CODE REVIEW

File: syntaxes/clarity.yaml [hunk 1]
```diff
@@ -1,184 +1,186 @@
...
-      name: constant.character.escape.clarity
-    - include: '#sexp'
-    - include: '#illegal'
+            \)    # close parens
+        captures:
+          "1":
+            name: keyword.control.clarity
+          "2":
+            name: entity.name.function.clarity
+          "3":
```
Comment (reedrosenbluth) [inline syntaxes/clarity.yaml:140]: this seems to fix an issue where only the last function arg would get highlighted

File: syntaxes/clarity.yaml [hunk 2]
```diff
@@ -1,184 +1,186 @@
...
+            (?<=(\s|\()) # preceded by space or (
+            ( is-eq|is-some|is-none|is-ok|is-err )
+            (?=(\s|\()) # followed by space or (
+        name: support.function.boolean-test.clarity
+      - match: >
+          (?x)
+            (?<=(\s|\()) # preceded by space or (
+            ( var-set|map-set|map-delete|map-insert|
+              ft-transfer\?|nft-transfer\?|nft-mint\?|ft-mint\?|nft-get-owner\?|ft-get-balance\?|
+              contract-call\?|stx-burn\?|stx-transfer\?|stx-get-balance)
```
Comment (reedrosenbluth) [inline syntaxes/clarity.yaml]: new functions

File: syntaxes/clarity.yaml [hunk 3]
```diff
@@ -1,184 +1,186 @@
...
+      - match: >
+          (?x)
+            (?<=(\s|\()) # preceded by space or (
+            ( as-contract|contract-caller|tx-sender|block-height|at-block|get-block-info\? )
+            (?=(\s|\()) # followed by space or (
+        name: support.function.sender-context.clarity
+      - match: >
+          (?x)
+            (?<=(\s|\()) # preceded by space or (
+            ( >=?|<=?|[*/+-]|mod|pow|log2 )
```
Comment (reedrosenbluth) [inline syntaxes/clarity.yaml:92]: new function

File: syntaxes/clarity.yaml [hunk 4]
```diff
@@ -1,184 +1,186 @@
...
-    - match: \'[0123456789ABCDEFGHJKMNPQRSTVWXYZ]{28,41}
-      name: constant.numeric.clarity
+      - match: (true|false)
+        name: constant.language.boolean.clarity
+      - match: (?<=[\(\s])((#e|#i)?[0-9]+(\.[0-9]+)?|(#x)[0-9a-fA-F]+|(#o)[0-7]+|(#b)[01]+)(?=[\s;()'",\[\]])
+        name: constant.numeric.clarity
+      - match: \'[0123456789ABCDEFGHJKMNPQRSTVWXYZ]{28,41}
+        name: constant.numeric.clarity
+      - match: u[0123456789]{1,}
+        name: constant.numeric.clarity
```
Comment (reedrosenbluth) [inline syntaxes/clarity.yaml:52]: match `uints`

**Comment by hstove** [comment]
Is `use-trait` properly handled? Especially the bracket syntax.

```clar
(use-trait trait-name STXX.contract.trait)

(define-public (somefun (contract <trait-name>)))
```

**Review by aulneau — APPROVED** [review]

**Review by kantai — APPROVED** [review]

**Comment by reedrosenbluth** [comment]
@hstove just added highlighting for `use-trait`, but I think we should create an issue for the bracket syntax.

**Pull request merged**